File: dojo_9/java/src/main/java/com/mercadolibre/dojos/Transporte.java
package com.mercadolibre.dojos;

public abstract class Transporte {
    private final Precio precioPorKM;

    public Transporte(Precio precioPorKM) {
        this.precioPorKM = precioPorKM;
    }

    public Precio precio(Lugar origen, Lugar destino) {
        return precioPorKM.multiply(origen.distancia(destino));
    }
}

File: dojo_kotlin/app/src/main/java/com/nelsito/kotlindojo9/Transporte.kt
package com.nelsito.kotlindojo9

/**
 * Created by npuglisi on 12/12/17.
 */
abstract class Transporte(val precioPorKM: Precio) {
    fun precio(origen: Lugar, destino: Lugar): Precio {
        return precioPorKM.multiply(origen.distancia(destino))
    }
}